File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        int i = 0;
        int j = n - 1;
        int count = 0;
        while(i < m) {
            while(j >= 0 && grid[i][j] < 0)
                j--;
            count += n - j - 1;
            i++;
        }
        return count;
    }
};

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351. Count Negative Numbers in a Sorted Matrix.go
func countNegatives(grid [][]int) int {
    m := len(grid)
    n := len(grid[0])
    j := n - 1
    count := 0
    for i := 0; i < m; i++ {
        for j >= 0 && grid[i][j] < 0 {j--}
        count += n - j - 1
    }
    return count
}
